core/trino-main/src/test/java/io/trino/execution/BaseRoleTaskTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.trino.client.NodeVersion;
+import io.trino.connector.MockConnectorFactory;
+import io.trino.connector.MockConnectorPlugin;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.Metadata;
+import io.trino.security.AccessControl;
+import io.trino.spi.resourcegroups.ResourceGroupId;
+import io.trino.spi.security.Identity;
+import io.trino.spi.security.RoleGrant;
+import io.trino.spi.security.TrinoPrincipal;
+import io.trino.sql.parser.SqlParser;
+import io.trino.sql.tree.Statement;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.StandaloneQueryRunner;
+import io.trino.transaction.TransactionManager;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+
+import java.net.URI;
+import java.util.Optional;
+import java.util.concurrent.ExecutorService;
+
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.trino.SessionTestUtils.TEST_SESSION;
+import static io.trino.execution.querystats.PlanOptimizersStatsCollector.createPlanOptimizersStatsCollector;
+import static io.trino.spi.security.PrincipalType.USER;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+/**
+ * Base class for various *RoleTask test classes.  Distinct from {@link BaseDataDefinitionTaskTest} due to different catalogs and metadata.
+ */
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class BaseRoleTaskTest
```

Comment:
Don't use inheritance for this. It makes it harder to maintain the tests, as the subclasses are intimately coupled to the details of the parent. Over time, as some of the tests evolve, the parent becomes a kitchen sink of functionality.

Reply:
What would be a good way to factor out the boilerplate and make new tests easy to add?  I thought about putting the DropRole tests and SetRole tests into the same class, but I like the regularity of `Test<class name>Task` naming.

I could, of course, copy-paste the boilerplate into `TestDropRoleTask`, but that felt unsatisfying.

Reply:
If all the role tests require similar setup, it's an ok compromise to have them in the same class (e.g., TestRoleTasks). They may require different setup and boilerplate, though, at which point they'd need to get split into separate classes with their relevant setup code.

Reply:
> it's an ok compromise to have them in the same class (e.g., TestRoleTasks).

Done.

> They may require different setup and boilerplate, though, at which point they'd need to get split into separate classes with their relevant setup code.

ACK